src/node_crypto.cc

```
@@ -2749,6 +2753,7 @@ static bool IsValidGCMTagLength(unsigned int tag_len) {
 bool CipherBase::InitAuthenticated(const char* cipher_type, int iv_len,
                                    unsigned int auth_tag_len) {
   CHECK(IsAuthenticatedMode());
+  MarkPopErrorOnReturn mark_pop_error_on_return;
```

For my own understanding: `MarkPopErrorOnReturn` basically means that OpenSSL errors inside this scope are swallowed/ignored?

---

Exactly, all errors that occur in this scope are dropped once the deconstructor is called (unless they have been handled) :-)

---

And now that we're handling the errors manually, Node will be throwing an error instead of just crashing, I suppose?

---

And because we're manually handling the errors now, it's better because we'll throw an error that you could catch and what not instead of just crashing, right?

---

It won't throw more errors than before (except for `::Init`), it will just make sure that *if* it throws, it also clears the internal error queue of OpenSSL.